spec/protect/text_processor_spec.rb

```diff
@@ -55,8 +55,8 @@
           ],
           tokenizer: { kind: :standard }
         })
-      result = tokenizer.perform("This is an example of an ngram filter")
-      expect(result).to eq(["thi", "his", "exa", "xam", "amp", "mpl", "ple", "ngr", "gra", "ram", "fil", "ilt", "lte", "ter"])
+      result = tokenizer.perform("Example filter")
+      expect(result).to eq(["exa", "xam", "amp", "mpl", "ple", "exa", "exam", "examp", "exampl", "example", "fil", "ilt", "lte", "ter", "fil", "filt", "filte", "filter"])
```

Comment:
From Dan's example:
> In other words, the query term “coffee” would result in the output array:
> `["cof", "off", "ffe", "fee", "coff", "ffee", "coffe", "offee", "coffee"]`

Having a look at the above list:
- There are duplicates; is this expected? (eg. `exa` and `fil` are appearing twice.)
- I'd also expect `xamp`, `xampl`, `xample`, `ampl`, `ample` and `mple` to appear for the `example` word.

(Not a criticism, as I'm not sure I have the correct understanding of this.)

Reply:
... but on the other hand, https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-edgengram-tokenizer.html#_example_configuration_7 only has those prefixes.

uhhhh help please, @coderdan

Reply:
@damncabbage "edge" ngrams only do prefixes (or suffixes, I guess) of varying lengths. A plain ngram is not limited to prefixes only.

Reply:
@damncabbage ah I think i've misinterpreted the example @coderdan gave in the ticket. I read that as showing a combination of ngrams and edge ngrams, but look's like I might be wrong.

Reply:
@fi re: what @damncabbage mentioned about duplicates - I wouldn't expect to see duplicates in the array.

Reply:
Thinking about the duplicates some more - if a `Set` is returned (instead of an array) we won't even be able to represent duplicates at all. It matches the semantics better.